maven/impl-maven-embedded/src/test/java/org/jboss/shrinkwrap/resolver/impl/maven/embedded/MavenExtractionTestCase.java

```diff
@@ -0,0 +1,77 @@
+package org.jboss.shrinkwrap.resolver.impl.maven.embedded;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.TimeUnit;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import org.apache.commons.io.FileUtils;
+import org.jboss.shrinkwrap.resolver.api.maven.embedded.EmbeddedMaven;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.contrib.java.lang.system.SystemOutRule;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.DistributionStageImpl.MAVEN_TARGET_DIR;
+import static org.jboss.shrinkwrap.resolver.impl.maven.embedded.Utils.pathToJarSamplePom;
+
+public class MavenExtractionTestCase {
+
+   @Rule
+   public final SystemOutRule systemOutRule = new SystemOutRule().enableLog();
+   private File targetMavenDir = new File(MAVEN_TARGET_DIR);
+
+   @Before
+   public void cleanup() throws IOException {
+      FileUtils.deleteDirectory(targetMavenDir);
+      targetMavenDir.mkdirs();
+   }
+
+   @Test
+   public void testDownloadDefaultMavenAndExtractUsingMultipleThreads() throws IOException, InterruptedException {
+      // download once
+      final URL mavenDistribution =
+         new URL("http://redrockdigimark.com/apachemirror/maven/maven-3/3.5.2/binaries/apache-maven-3.5.2-bin.tar.gz");
```

Comment:
who is the `http://redrockdigimark.com/`? it's just some apache mirror probably - it's better to use the apache link instead: https://archive.apache.org/dist/maven/maven-3/3.5.2/